nexus/src/app/sagas/region_replacement_drive.rs

```diff
@@ -0,0 +1,1521 @@
+                    vmm.id, vmm_id,
+                )));
+            }
+
+            info!(
+                log,
+                "sending replacement request for disk volume to propolis {vmm_id}";
+                "region replacement id" => %params.request.id,
+                "disk id" => ?disk.id(),
+                "volume id" => ?disk.volume_id,
+            );
+
+            // Start (or poll) the replacement
+            let result = client
+                .instance_issue_crucible_vcr_request()
+                .id(disk.id())
+                .body(
+                    propolis_client::types::InstanceVcrReplace {
+                        name: disk.name().to_string(),
+                        vcr_json: disk_new_volume_vcr.to_string(),
+                    }
+                )
+                .send()
+                .await
+                .map_err(|e| match e {
+                    propolis_client::Error::ErrorResponse(
+                        rv,
+                    ) => {
+                        ActionError::action_failed(rv.message.clone())
+                    }
+
+                    _ => {
+                        ActionError::action_failed(
+                            format!(
+                                "unexpected failure during `instance_issue_crucible_vcr_request`: {e}",
+                            )
+                        )
+                    }
+                })?;
+
+            let replace_result = result.into_inner();
+
+            info!(
+                log,
+                "saw replace result {replace_result:?}";
+                "region replacement id" => %params.request.id,
+                "disk id" => ?disk.id(),
+                "volume id" => ?disk.volume_id,
+            );
+
+            let replacement_done = match &replace_result {
+                ReplaceResult::Started => {
+                    // This drive saga's call just started the replacement
+                    false
+                }
+
+                ReplaceResult::StartedAlready => {
+                    // A previous drive saga's call started the replacement, but
+                    // it's not done yet.
+                    false
+                }
+
+                ReplaceResult::CompletedAlready => {
+                    // It's done! We see this if the same propolis that received
+                    // the original replace request started and finished the
+                    // live repair.
+                    true
+                }
+
+                ReplaceResult::VcrMatches => {
+                    // If this propolis booted after the volume construction
+                    // request was modified but before all the regions were
+                    // reconciled, then `VcrMatches` will be seen as a result of
+                    // `target_replace`: the new propolis will have received the
+                    // updated VCR when it was created.
+                    //
```

Comment:
Did you test this path?  Replacing a downstairs after activation was requested but before
the reconciliation finished?  I've not done it, so I'm not sure what it will tell you when
you ask the upstairs to replace a downstairs.

Reply:
This is covered in the test case `a disk needing repair is attached to a instance that is stopped mid-repair, then started in the middle of the pantry's repair`, yeah